File: qt/gen_qtextboundaryfinder.cpp
// +build ignore

#include <QChar>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QTextBoundaryFinder>
#include <qtextboundaryfinder.h>
#include "gen_qtextboundaryfinder.h"

#ifndef _Bool
#define _Bool bool
#endif

void _GUID_Delete(_GUID* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<_GUID*>( self );
	} else {
		delete self;
	}
}

void type_info_Delete(type_info* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<type_info*>( self );
	} else {
		delete self;
	}
}

QTextBoundaryFinder* QTextBoundaryFinder_new() {
	return new QTextBoundaryFinder();
}

QTextBoundaryFinder* QTextBoundaryFinder_new2(QTextBoundaryFinder* other) {
	return new QTextBoundaryFinder(*other);
}

QTextBoundaryFinder* QTextBoundaryFinder_new3(BoundaryType typeVal, struct miqt_string stringVal) {
	QString stringVal_QString = QString::fromUtf8(stringVal.data, stringVal.len);
	return new QTextBoundaryFinder(typeVal, stringVal_QString);
}

QTextBoundaryFinder* QTextBoundaryFinder_new4(BoundaryType typeVal, QChar* chars, ptrdiff_t length) {
	return new QTextBoundaryFinder(typeVal, chars, (qsizetype)(length));
}

QTextBoundaryFinder* QTextBoundaryFinder_new5(BoundaryType typeVal, QChar* chars, ptrdiff_t length, unsigned char* buffer) {
	return new QTextBoundaryFinder(typeVal, chars, (qsizetype)(length), static_cast<unsigned char*>(buffer));
}

QTextBoundaryFinder* QTextBoundaryFinder_new6(BoundaryType typeVal, QChar* chars, ptrdiff_t length, unsigned char* buffer, ptrdiff_t bufferSize) {
	return new QTextBoundaryFinder(typeVal, chars, (qsizetype)(length), static_cast<unsigned char*>(buffer), (qsizetype)(bufferSize));
}

void QTextBoundaryFinder_OperatorAssign(QTextBoundaryFinder* self, QTextBoundaryFinder* other) {
	self->operator=(*other);
}

bool QTextBoundaryFinder_IsValid(const QTextBoundaryFinder* self) {
	return self->isValid();
}

BoundaryType QTextBoundaryFinder_Type(const QTextBoundaryFinder* self) {
	return self->type();
}

struct miqt_string QTextBoundaryFinder_String(const QTextBoundaryFinder* self) {
	QString _ret = self->string();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

void QTextBoundaryFinder_ToStart(QTextBoundaryFinder* self) {
	self->toStart();
}

void QTextBoundaryFinder_ToEnd(QTextBoundaryFinder* self) {
	self->toEnd();
}

ptrdiff_t QTextBoundaryFinder_Position(const QTextBoundaryFinder* self) {
	qsizetype _ret = self->position();
	return static_cast<ptrdiff_t>(_ret);
}

void QTextBoundaryFinder_SetPosition(QTextBoundaryFinder* self, ptrdiff_t position) {
	self->setPosition((qsizetype)(position));
}

ptrdiff_t QTextBoundaryFinder_ToNextBoundary(QTextBoundaryFinder* self) {
	qsizetype _ret = self->toNextBoundary();
	return static_cast<ptrdiff_t>(_ret);
}

ptrdiff_t QTextBoundaryFinder_ToPreviousBoundary(QTextBoundaryFinder* self) {
	qsizetype _ret = self->toPreviousBoundary();
	return static_cast<ptrdiff_t>(_ret);
}

bool QTextBoundaryFinder_IsAtBoundary(const QTextBoundaryFinder* self) {
	return self->isAtBoundary();
}

BoundaryReasons QTextBoundaryFinder_BoundaryReasons(const QTextBoundaryFinder* self) {
	return self->boundaryReasons();
}

void QTextBoundaryFinder_Delete(QTextBoundaryFinder* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QTextBoundaryFinder*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qtextboundaryfinder.go
package qt

import (
	"unsafe"
)

type QTextBoundaryFinder__BoundaryType int

const (
	QTextBoundaryFinder__Grapheme QTextBoundaryFinder__BoundaryType = 0
	QTextBoundaryFinder__Word     QTextBoundaryFinder__BoundaryType = 1
	QTextBoundaryFinder__Sentence QTextBoundaryFinder__BoundaryType = 2
	QTextBoundaryFinder__Line     QTextBoundaryFinder__BoundaryType = 3
)

type QTextBoundaryFinder__BoundaryReason int

const (
	QTextBoundaryFinder__NotAtBoundary    QTextBoundaryFinder__BoundaryReason = 0
	QTextBoundaryFinder__BreakOpportunity QTextBoundaryFinder__BoundaryReason = 31
	QTextBoundaryFinder__StartOfItem      QTextBoundaryFinder__BoundaryReason = 32
	QTextBoundaryFinder__EndOfItem        QTextBoundaryFinder__BoundaryReason = 64
	QTextBoundaryFinder__MandatoryBreak   QTextBoundaryFinder__BoundaryReason = 128
	QTextBoundaryFinder__SoftHyphen       QTextBoundaryFinder__BoundaryReason = 256
)

type QTextBoundaryFinder struct {
	h          uintptr
	isSubclass bool
}

// NewQTextBoundaryFinder constructs a new QTextBoundaryFinder object.
func NewQTextBoundaryFinder() *QTextBoundaryFinder {
	ret := newQTextBoundaryFinder(QTextBoundaryFinder_new())
	ret.isSubclass = true
	return ret
}

// NewQTextBoundaryFinder2 constructs a new QTextBoundaryFinder object.
func NewQTextBoundaryFinder2(other *QTextBoundaryFinder) *QTextBoundaryFinder {
	ret := newQTextBoundaryFinder(QTextBoundaryFinder_new2(other.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQTextBoundaryFinder3 constructs a new QTextBoundaryFinder object.
func NewQTextBoundaryFinder3(typeVal BoundaryType, stringVal string) *QTextBoundaryFinder {
	stringVal_ms := struct_miqt_string{}
	stringVal_ms.data = CString(stringVal)
	stringVal_ms.len = size_t(len(stringVal))
	defer free(unsafe.Pointer(stringVal_ms.data))

	ret := newQTextBoundaryFinder(QTextBoundaryFinder_new3(typeVal, stringVal_ms))
	ret.isSubclass = true
	return ret
}

// NewQTextBoundaryFinder4 constructs a new QTextBoundaryFinder object.
func NewQTextBoundaryFinder4(typeVal BoundaryType, chars *QChar, length int64) *QTextBoundaryFinder {
	ret := newQTextBoundaryFinder(QTextBoundaryFinder_new4(typeVal, chars.cPointer(), (ptrdiff_t)(length)))
	ret.isSubclass = true
	return ret
}

// NewQTextBoundaryFinder5 constructs a new QTextBoundaryFinder object.
func NewQTextBoundaryFinder5(typeVal BoundaryType, chars *QChar, length int64, buffer *byte) *QTextBoundaryFinder {
	ret := newQTextBoundaryFinder(QTextBoundaryFinder_new5(typeVal, chars.cPointer(), (ptrdiff_t)(length), (*uchar)(unsafe.Pointer(buffer))))
	ret.isSubclass = true
	return ret
}

// NewQTextBoundaryFinder6 constructs a new QTextBoundaryFinder object.
func NewQTextBoundaryFinder6(typeVal BoundaryType, chars *QChar, length int64, buffer *byte, bufferSize int64) *QTextBoundaryFinder {
	ret := newQTextBoundaryFinder(QTextBoundaryFinder_new6(typeVal, chars.cPointer(), (ptrdiff_t)(length), (*uchar)(unsafe.Pointer(buffer)), (ptrdiff_t)(bufferSize)))
	ret.isSubclass = true
	return ret
}

func (this *QTextBoundaryFinder) OperatorAssign(other *QTextBoundaryFinder) {
	QTextBoundaryFinder_OperatorAssign(this.h, other.cPointer())
}

func (this *QTextBoundaryFinder) IsValid() bool {
	return (bool)(QTextBoundaryFinder_IsValid(this.h))
}

func (this *QTextBoundaryFinder) Type() BoundaryType {
	xxxxxxxxx
}

func (this *QTextBoundaryFinder) String() string {
	var _ms struct_miqt_string = QTextBoundaryFinder_String(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QTextBoundaryFinder) ToStart() {
	QTextBoundaryFinder_ToStart(this.h)
}

func (this *QTextBoundaryFinder) ToEnd() {
	QTextBoundaryFinder_ToEnd(this.h)
}

func (this *QTextBoundaryFinder) Position() int64 {
	return (int64)(QTextBoundaryFinder_Position(this.h))
}

func (this *QTextBoundaryFinder) SetPosition(position int64) {
	QTextBoundaryFinder_SetPosition(this.h, (ptrdiff_t)(position))
}

func (this *QTextBoundaryFinder) ToNextBoundary() int64 {
	return (int64)(QTextBoundaryFinder_ToNextBoundary(this.h))
}

func (this *QTextBoundaryFinder) ToPreviousBoundary() int64 {
	return (int64)(QTextBoundaryFinder_ToPreviousBoundary(this.h))
}

func (this *QTextBoundaryFinder) IsAtBoundary() bool {
	return (bool)(QTextBoundaryFinder_IsAtBoundary(this.h))
}

func (this *QTextBoundaryFinder) BoundaryReasons() BoundaryReasons {
	xxxxxxxxx
}
